sdk/search/azure-search-documents/azure/search/documents/_internal/_index_documents_batch.py

```diff
@@ -98,22 +105,75 @@ def add_merge_or_upload_actions(self, *documents):
          index. May be a single list of documents, or documents as individual
          parameters.
         :type documents: dict or list[dict]
+        :rtype: List[IndexAction]
         """
-        self._extend_batch(flatten_args(documents), "mergeOrUpload")
+        return self._extend_batch(flatten_args(documents), "mergeOrUpload")
 
     @property
     def actions(self):
         # type: () -> List[IndexAction]
-        """The list of currently configured index actions.
+        """The list of currently index actions to index.
 
         :rtype: List[IndexAction]
         """
         return list(self._actions)
 
+    @property
+    def succeeded_actions(self):
+        # type: () -> List[IndexAction]
+        """The list of currently succeeded index actions.
+
+        :rtype: List[IndexAction]
+        """
+        return list(self._succeeded_actions)
```

Comment:
Properties generally return the same instance every time. Here we are creating a new list every time.

Reply:
This is my intension.  I don't want succeeded_actions to be modified outside of this class.

Reply:
Then I would make it into a function rather than a property. Making it a property implies that modifications to the returned list are persisted...